lib/vanagon/platform/win.rb

```diff
@@ -0,0 +1,118 @@
+      def generate_package(project)
+        target_dir = project.repo ? output_dir(project.repo) : output_dir
+        ["mkdir -p output/#{target_dir}",
+        "mkdir -p $(tempdir)/#{project.name}-#{project.version}",
+        "cp #{project.name}-#{project.version}.tar.gz $(tempdir)/#{project.name}_#{project.version}.orig.tar.gz",
+        "cat file-list >> windows/install",
+        "cp -pr windows $(tempdir)/#{project.name}-#{project.version}",
+        "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/#{project.name}-#{project.version}' --strip-components 1 -xf -",
+        "sed -i 's/\ /?/g' $(tempdir)/#{project.name}-#{project.version}/windows/install",
+        "(cd $(tempdir)/#{project.name}-#{project.version}; debuild --no-lintian -uc -us)",
+        "cp $(tempdir)/*.win ./output/#{target_dir}"]
+      end
+
+      # Method to generate the files required to build a windows package for the project
+      #
+      # @param workdir [String] working directory to stage the evaluated templates in
+      # @param name [String] name of the project
+      # @param binding [Binding] binding to use in evaluating the packaging templates
+      def generate_packaging_artifacts(workdir, name, binding)
+        win_dir = File.join(workdir, "windows")
+        FileUtils.mkdir_p(win_dir)
+
+        # Lots of templates here
+        ["changelog", "conffiles", "control", "docs", "dirs", "install", "preinst", "postinst", "postrm", "prerm", "rules"].each do |win_file|
+          erb_file(File.join(VANAGON_ROOT, "templates/win/#{win_file}.erb"), File.join(win_dir, win_file), false, { :binding => binding })
+        end
+
+        # These could be templates, but their content is static, so that seems weird.
+        File.open(File.join(win_dir, "compat"), "w") { |f| f.puts("7") }
+        FileUtils.mkdir_p(File.join(win_dir, "source"))
+        File.open(File.join(win_dir, "source", "format"), "w") { |f| f.puts("3.0 (quilt)") }
+      end
+
+      # Method to derive the package name for the project
+      #
+      # @param project [Vanagon::Project] project to name
+      # @return [String] name of the windows package for this project
+      def package_name(project)
+        "#{project.name}_#{project.version}-#{project.release}#{@codename}_#{project.noarch ? 'all' : @architecture}.win"
+      end
+
+      # Get the expected output dir for the windows packages. This allows us to
+      # use some standard tools to ship internally.
+      #
+      # @return [String] relative path to where windows packages should be staged
+      def output_dir(target_repo = "")
+        File.join("win", @codename, target_repo)
+      end
+
+      # Returns the string to add a target repo to the platforms' provisioning
+      #
+      # @param definition [URI] A URI to a win or list file
+      # @return [String] The command to add the repo target to the system
+      def add_repo_target(definition)
```

Comment:
We shouldn't need any of this method yet, so we should be able to safely delete it entirely